src/schoology-extractor/schoology_extractor/main.py

```diff
@@ -36,29 +36,40 @@
 
 export_data.to_csv(users_list, f"{schoology_output_path}/users.csv")
 
-
 # export sections
-sections_list = []
-for section_id in sections_id_array:
-    sections_list.append(request_client.get_section_by_id(section_id))
 
+# first we need to get a list of courses
+courses_response = request_client.get_courses()
+courses_list = []
+while True:
+    courses_list = courses_list + courses_response.current_page_items
+    if courses_response.get_next_page() is None:
+        break
+
+# now we can get a list of sections
+course_ids = map(lambda x: x["id"], courses_list)
+sections_list = request_client.get_section_by_course_ids(list(course_ids))
 export_data.to_csv(sections_list,
                    f"{schoology_output_path}/sections.csv")
 
 
 # export assigments
-export_data.to_csv(request_client.get_assignments_by_section_ids(sections_id_array),
+section_ids = map(lambda x: x["id"], sections_list)
```

Comment:
This code seems fine, but I'm curious why you chose `map` instead of writing a list comprehension?

Reply:
No specific reason... Normally I use list comprehension only when I need to filter something, but any of them would be fine.

Reply:
It is not worth changing this code. In general I recommend sticking with list comprehensions even when not filtering something, because it is strange to see a mixture of the two and slows down my reading of the code.

Interesting performance comparison: https://stackoverflow.com/a/1247490/30384